src/main/java/org/rumbledb/server/RumbleHandler.java

```diff
@@ -0,0 +1,203 @@
+        }
+        return args;
+    }
+
+    @Override
+    public void handle(HttpExchange exchange) throws IOException {
+        try {
+            URI uri = exchange.getRequestURI();
+            String queryString = uri.getQuery();
+            String args[] = getCLIArguments(queryString);
+
+            RumbleRuntimeConfiguration configuration = new RumbleRuntimeConfiguration(args);
+            SparkSessionManager.COLLECT_ITEM_LIMIT = configuration.getResultSizeCap();
+
+            JsoniqQueryExecutor translator = new JsoniqQueryExecutor(configuration);
+            List<Item> items;
+            if (configuration.getQueryPath() != null) {
+                if (configuration.getOutputPath() != null && !exchange.getRequestMethod().equals("POST")) {
+                    throw new CliException(
+                            "The POST method must be used if specifying an output path, as this has side effects."
+                    );
+                }
+                items = translator.runQuery(
+                    configuration.getQueryPath(),
+                    configuration.getOutputPath()
+                );
+            } else {
+                if (configuration.getOutputPath() != null && !exchange.getRequestMethod().equals("POST")) {
+                    throw new CliException(
+                            "The POST method must be used if specifying an output path, as this has side effects."
+                    );
+                }
+                if (!exchange.getRequestMethod().equals("GET") && !exchange.getRequestMethod().equals("POST")) {
+                    throw new CliException("Only the GET and POST methods are supported.");
+                }
+                InputStreamReader r = new InputStreamReader(exchange.getRequestBody());
+                BufferedReader r2 = new BufferedReader(r);
+                StringBuffer sb = new StringBuffer();
+                String s;
+                while ((s = r2.readLine()) != null) {
+                    sb.append(s);
+                }
+                String JSONiqQuery = sb.toString();
+                items = translator.runInteractive(JSONiqQuery);
+            }
+
+            Item output = ItemFactory.getInstance().createObjectItem();
+            if (items != null) {
+                Item values = ItemFactory.getInstance().createArrayItem(items);
+                output.putItemByKey("values", values);
+            }
+            output.putItemByKey("status", ItemFactory.getInstance().createIntegerItem(200));
+            if (configuration.getOutputPath() != null) {
+                output.putItemByKey(
+                    "output-path",
+                    ItemFactory.getInstance().createStringItem(configuration.getOutputPath())
+                );
+            }
+            if (configuration.getLogPath() != null) {
+                output.putItemByKey("log-path", ItemFactory.getInstance().createStringItem(configuration.getLogPath()));
+            }
```

Comment:
I would recommend extracting lines 121-135 to a function such as `prepareOutput(...)` for increased readability.

Reply:
DONE